M2/Macaulay2/packages/IncidenceCorrespondenceCohomology.m2

```diff
@@ -0,0 +1,2105 @@
+--        such that \delta_c appears (with some shift) in the product \delta_a indexed by a in L
+
+recSLP = (p,L) -> (
+    if #L == 0 then new HashTable from {1 => 0} else
+    (
+       H := recSLP(p,drop(L,1));
+       if H === false then false else SLPair(p,L#0,H)
+       )
+)
+
+--auxiliary function, to check if  k[x,y]/(x^a y^b) using Theorem 3.2 [Nicklasson17]
+--input:  prime p and  integers a and b with a,b>=p and p>2
+--output:  true if  k[x,y]/(x^a y^b) has the SLP in characteristic p, false otherwise
+--requres p>2 and a,b>p-1
+
+auxSLP2= (p,a,b)->(
+    a0:=a%p;
+    b0:=b%p;
+    if a0==(p-1)/2 or a0==(p+1)/2 then(
+	if b0==(p-1)/2 or b0==(p+1)/2 then(
+	    k1:= boundq(p,a)-1;
+	    k2:= boundq(p,b)-1;
+	    k:= min {k1,k2};
+		-- then a=ak1*p^k1+.... a1*p+a0, we check that aj=(p-1)/2 for j=1,...,k-1
+		if (a%p^k)-a0==(p^k -p)/2 then( 
+		    -- similarly  b=bk2*p^k2+.... b1*p+a0, we check that bj=(p-1)/2 for j=1,...,k-1
+		    if (b%p^k)-b0== (p^k -p)/2 then(
+			ak:= (a%p^(k+1))//p^k;
+			bk:= (b%p^(k+1))//p^k;
+			if ak+bk<p then (
+			    if k1==k2 then true 
+			    else  if( k1<k2 and ak<= bk) or (k2<k1 and bk<= ak) then true
+			    else false)---if any of the condition above fails return false
+		      	else false)                   
+                    else false)               
+                else false)             
+            else false)         
+        else false)
+
+
+--User function: This function checks if the monomial complete intersection (ZZ/p)[x_1,...,x_n]/(x_1^{a_1},...,x_n^{a_n})  has the Strong Lefschetz Property (SLP) in characteristic p 
+--This function uses the classification presented in Theorem 3.8 [Lundqvist, Nicklasson'18], or, when the option UseConjecture is set to be true, the Han-Monsky multiplication adopting Conjecture 4.1 [KMRR,25]
+--input: (p, L) where p is prime and L={a_1,a_2 ... a_n} is a list of integers 
+--output: true if  k[x_1,...,x_n]/(x_1^{a_1}, …, x_n^{a_n}) has the SLP in characteristic p, false otherwise
+
+
+hasSLP = method(Options => {UseConjecture => false})
+
+hasSLP(ZZ, List) := opts -> (p, L) -> (
+    n:=#L;
+    if n <= 1 or p == 0 then true else (
+    	if sum L - n < p then true 
+	else (
+    	    if not(opts.UseConjecture) then ( 
+		if p==2 then ( 
+		    if n==2 and ((L#0==2 and  L#1%2==1)  or (L#1==2 and  L#0%2==1) or 
+			(L#0==3 and  L#1%4==2)  or (L#0==3 and  L#1%4==2))   then true	 -- by Corollary 6.3. [Cook'11]  
+		    else false)-- by Corollary 4.8  [Cook'11] if n>2 and p=2 we don't have the SLP
+		else( L1:=select(L, x -> x>=p);-- list the a_j bigger then p
+		    if #L1!=1 then  ( 
+			if #L==2 and #L1==2 then (return auxSLP2(p, L#0, L#1);)--if this is the case when n=2, p>2 and both a_1,a_2>=p, so we use auxSLP
+			else  false)-- if n>2 it is necessary that exactly one a_j is bigger then p
+		    else (
+			d:=L1#0;
+			d0:=d%p;
+			if sum L -n-(d-1)<=min{p-d0,d0} then true 
+			else  false
```

Comment:
In general, `if x then true else false` can be simplified down to just `x`.

Reply:
This is still an issue.  The suggestion is to replace these two lines with just:

`sum L -n-(d-1)<=min{p-d0,d0}` (i.e., remove the `if` and `then true else false`).